### PR TITLE
Dominate & Abyssal Gaze are now a pointed spells

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR makes Dominate and Abyssal Gaze a pointed spells, which means they will use direct mouse guidance to find their targets instead of using a clumsy input menu. 

I have also done some improvements and fixes to the pointed spells in general, like cutting redundant code regarding can_target, making sure anti magic will waste spell properly and lastly, added small aim assist to the pointed spells - If there is a single valid target on a turf user has clicked on, the spell will be automatically redirected to the valid target, as otherwise it could sometimes be a little hard to precisely click on the target mob sprite directly.

## Why It's Good For The Game

Better spell control.

## Changelog
:cl: Arkatos
add: Dominate is now a cursor-guided spell.
add: Abyssal Gaze is now a cursor-guided spell.
tweak: Pointed spells now have aim assist. If there is a single valid target on a turf user has clicked on, the spell will be automatically redirected to the valid target.
/:cl:

### CODE REVIEW

**Comment by Tharcoonvagh** [comment]
arent those shadowling abilities? whaaaaaat

**Comment by ShibaInuLord** [comment]
they're cult construct abilities, shadowlings dont have any aside from the jaunt.

**Comment by Tharcoonvagh** [comment]
i meant actual shadowlings but thanks anyways

File: code/modules/spells/spell_types/construct_spells.dm [hunk 1]
```diff
@@ -284,14 +271,22 @@
 			to_chat(user, "<span class='warning'>Target is not a lesser creature!</span>")
 		return FALSE
 
-	var/mob/living/simple_animal/animal = target
-	if(animal.stat)
+	var/mob/living/simple_animal/S = target
+	if(S.ckey)
```
Comment (Rohesie) [inline code/modules/spells/spell_types/construct_spells.dm:275]:
```suggestion
	if(S.mind)
```

**Comment by Rohesie** [comment]
This looks ready, save for that change request.
Is it properly tested locally?

**Comment by Arkatos1** [comment]
Both abilities are well tested with no issues, but I have to recreate this PR so we can get requested changes on it, as this is on my old fork.

**Pull request closed without merging**